internal/router/controllers/user.go

```diff
@@ -267,6 +270,23 @@ func (c *UserController) runReadRequest(
 	return res, true
 }
 
+func (c *UserController) collectReadQueryMetric(r *http.Request, statement string, config formatter.FormatConfig) {
+	value := r.Context().Value(middlewares.ContextIPAddress)
+	ipAddress, ok := value.(string)
+	if ok && ipAddress != "" {
```

Comment:
Only collecting if IP address was detected. We can change this, to always collect if that's make more sense

Reply:
Sounds fine. I think this should always be the case. 
Maybe it's worth adding a warning log for the `else`?